source/fuzz/transformation_vector_shuffle.cpp

```diff
@@ -151,46 +155,51 @@ void TransformationVectorShuffle::Apply(
-      // This component is undefined, so move on - but first note that the
-      // overall shuffle result cannot be synonymous with any vector.
-      continue;
-    }
+  if (!transformation_context->GetFactManager()->IdIsIrrelevant(
+          message_.vector1()) &&
+      !transformation_context->GetFactManager()->IdIsIrrelevant(
+          message_.vector2())) {
+    for (uint32_t component_index = 0;
+         component_index < static_cast<uint32_t>(message_.component_size());
+         component_index++) {
+      uint32_t component = message_.component(component_index);
+      if (component == 0xFFFFFFFF) {
+        // This component is undefined, so move on - but first note that the
+        // overall shuffle result cannot be synonymous with any vector.
+        continue;
+      }
 
-    // This describes the element of the result vector associated with
-    // |component_index|.
-    protobufs::DataDescriptor descriptor_for_result_component =
-        MakeDataDescriptor(message_.fresh_id(), {component_index});
-
-    protobufs::DataDescriptor descriptor_for_source_component;
-
-    // Get a data descriptor for the component of the input vector to which
-    // |component| refers.
-    if (component <
-        GetVectorType(ir_context, message_.vector1())->element_count()) {
-      descriptor_for_source_component =
-          MakeDataDescriptor(message_.vector1(), {component});
-    } else {
-      auto index_into_vector_2 =
-          component -
-          GetVectorType(ir_context, message_.vector1())->element_count();
-      assert(
-          index_into_vector_2 <
-              GetVectorType(ir_context, message_.vector2())->element_count() &&
-          "Vector shuffle index is out of bounds.");
-      descriptor_for_source_component =
-          MakeDataDescriptor(message_.vector2(), {index_into_vector_2});
-    }
+      // This describes the element of the result vector associated with
+      // |component_index|.
+      protobufs::DataDescriptor descriptor_for_result_component =
+          MakeDataDescriptor(message_.fresh_id(), {component_index});
+
+      protobufs::DataDescriptor descriptor_for_source_component;
+
+      // Get a data descriptor for the component of the input vector to which
+      // |component| refers.
+      if (component <
+          GetVectorType(ir_context, message_.vector1())->element_count()) {
+        descriptor_for_source_component =
+            MakeDataDescriptor(message_.vector1(), {component});
+      } else {
+        auto index_into_vector_2 =
+            component -
+            GetVectorType(ir_context, message_.vector1())->element_count();
+        assert(index_into_vector_2 <
+                   GetVectorType(ir_context, message_.vector2())
+                       ->element_count() &&
+               "Vector shuffle index is out of bounds.");
+        descriptor_for_source_component =
+            MakeDataDescriptor(message_.vector2(), {index_into_vector_2});
```

Comment:
How about being more fine-grained here and doing the "is irrelevant" checks for vector 1 and vector 2 here?  That way if vector 1 is irrelevant but vector 2 is relevant we can still get facts for the parts of vector 2 that get shuffled into the result.

Reply:
Fixed in 5bea7f6.